web/web.go

```diff
@@ -96,7 +96,14 @@ func CheckBrowserCompatability(ua *user_agent.UserAgent) bool {
 }
 
 func root(c *api.Context, w http.ResponseWriter, r *http.Request) {
-	if !CheckBrowserCompatability(user_agent.New(r.UserAgent())) {
+	agentString := r.UserAgent()
+	ua := user_agent.New(agentString)
+
+	if strings.Contains(agentString, "Mattermost") {
+		l4g.Debug("Detected Browser: Mattermost App")
+	} else if ua.Mobile() {
+		l4g.Debug("Detected Browser: Mobile Browser")
+	} else if !CheckBrowserCompatability(ua) {
```

Comment:
While you're changing that to always print the browser, do you mind changing CheckBrowserCompatibility to take the user agent string and check for the app inside the function, so that it's all in one place? Having some cases in this if statement return and others not makes it a bit hard to follow the flow. You can also have CheckBrowserCompatibility print the name of the browser in each case